core/trino-main/src/main/java/io/trino/operator/aggregation/LongSumAggregation.java

```diff
@@ -13,53 +13,99 @@
  */
 package io.trino.operator.aggregation;
 
+import io.trino.annotation.UsedByGeneratedCode;
+import io.trino.operator.aggregation.state.NullableLongState;
 import io.trino.spi.block.BlockBuilder;
 import io.trino.spi.function.AggregationFunction;
 import io.trino.spi.function.AggregationState;
-import io.trino.spi.function.CombineFunction;
+import io.trino.spi.function.Decomposition;
 import io.trino.spi.function.InputFunction;
 import io.trino.spi.function.OutputFunction;
-import io.trino.spi.function.RemoveInputFunction;
 import io.trino.spi.function.SqlType;
-import io.trino.spi.type.BigintType;
+import io.trino.spi.function.WindowAccumulator;
+import io.trino.spi.function.WindowIndex;
 import io.trino.spi.type.StandardTypes;
 import io.trino.type.BigintOperators;
 
-@AggregationFunction("sum")
+import static io.trino.spi.type.BigintType.BIGINT;
+
+@AggregationFunction(value = "sum", windowAccumulator = LongSumAggregation.LongSumWindowAccumulator.class)
 public final class LongSumAggregation
 {
     private LongSumAggregation() {}
 
     @InputFunction
-    public static void sum(@AggregationState LongLongState state, @SqlType(StandardTypes.BIGINT) long value)
+    public static void sum(@AggregationState NullableLongState state, @SqlType(StandardTypes.BIGINT) long value)
     {
-        state.setFirst(state.getFirst() + 1);
-        state.setSecond(BigintOperators.add(state.getSecond(), value));
+        state.setNull(false);
```

Comment:
Cool! That will speed-up some basic queries

Reply:
@sopel39 There is a separate PR for this: https://github.com/trinodb/trino/pull/21349

This idea came from a conversation we had like 6 months ago, and finally got around to it 😄